### DISCUSSION TITLE
lint-staged with tsc-files cannot handle custom method types in cypress/support/e2e.ts

### DISCUSSION THREAD

**Discussion by radekBednarik** [cypress-io/cypress#23107] [Questions and Help]
Hello,
does somebody have a solution for this issue:
- I am using Cypress latest with Typescript
- I have setup a lint-staged pre-commit hook for the project
- I want to use typescript compiler in the pre-commit hook

Issue:
- I am using lint-stage coupled with tsc-files to have only commited files linted by the compiler. The problem is, that I have some custom commands with types declared in the `cypress/support/e2e.ts` file. No matter, what I try to do, the compiler will not get the types and I will get the error:

```shell
error TS2339: Property 'goto' does not exist on type 'cy & CyEventEmitter'.
```
Any ideas, how to get this work? I am currently stuck on this :(

**Comment by stavares843** [comment]
hello! 👋 

do you have the types declared on `support/index.js`?


also, does your `ts.config` file includes this?

```
{
  "include": [
    "integration/*.ts",
    "support/*.ts"
  ]
}
```
